### PR TITLE
Remove a couple of remaining deprecations

### PR DESCRIPTION
**Pull request by pirj**

### CODE REVIEW

File: lib/rspec/matchers/built_in/change.rb [hunk 1]
```diff
@@ -365,11 +363,6 @@ def initialize(matcher_name, receiver=nil, message=nil, &block)
           @receiver = receiver
           @message = message
           @value_proc = block
-          # TODO: temporary measure to mute warning of access to an initialized
-          # instance variable when a deprecated implicit block expectation
-          # syntax is used. This may be removed once `fail` is used, and the
-          # matcher never issues this warning.
-          @actual_after = UNDEFINED
```
Comment (JonRowe) [inline lib/rspec/matchers/built_in/change.rb:372]: Is this right?

Comment (pirj) [reply lib/rspec/matchers/built_in/change.rb:372]: Yes, previously we weren't initializing it https://github.com/rspec/rspec-expectations/blame/bd10f0cf3970932781efcd09b5e427877d16a6c2/lib/rspec/matchers/built_in/change.rb#L353

Comment (JonRowe) [reply lib/rspec/matchers/built_in/change.rb:372]: No I mean the removal in this PR

Comment (pirj) [reply lib/rspec/matchers/built_in/change.rb:372]: Oh. Yes. I've overlooked to remove this in #1285 after resolving conflicts with #1139.

I think this PR is the last change in Expectations to get it ready for the 4.0 release (apart from adding deprecation warnings to `main` and rebasing `4-0-dev` on top of it).

Sanity check, however, shows that nothing fails on https://github.com/rspec/rspec-expectations/tree/fe0c4b7f (#1139) even with the removed `@actual_after = UNDEFINED`. Can't recall the exact reason I've added it in the first place.

Comment (pirj) [reply lib/rspec/matchers/built_in/change.rb:372]: I'll extract this removal into a separate PR.

Comment (pirj) [reply lib/rspec/matchers/built_in/change.rb]: Done: #1289

**Review by JonRowe — APPROVED** [review]

**Pull request merged**